weave

```diff
@@ -804,6 +816,51 @@ dns_args() {
     fi
 }
 
+default_etc_hosts() {
+    name=$1
+    shift
+    extra_hosts=$@
```

Comment:
expansion of `$@` is rather special, so I suggest you drop this line and instead have `for host in "$@"...` below.